sdk/attestation/Azure.Security.Attestation/src/AttestationAdministrationClient.cs

```diff
@@ -136,26 +114,61 @@ public virtual AttestationResponse<string> GetPolicy(AttestationType attestation
+            => await GetPolicyInternalAsync(attestationType, true, cancellationToken).ConfigureAwait(false);
+
+        /// <summary>
+        /// Retrieves the attesttion policy for the specified <see cref="AttestationType"/>.
+        /// </summary>
+        /// <param name="attestationType"><see cref="AttestationType"/> to retrive.</param>
+        /// <param name="cancellationToken">Cancellation token used to cancel this operation.</param>
+        /// <param name="async">True if the call should be asynchronous.</param>
+        /// <returns>An <see cref="AttestationResponse{String}"/> with the policy for the specified attestation type.</returns>
+        /// <remarks>
+        /// This API returns the underlying attestation policy object stored in the attestation service for this <paramref name="attestationType"/>.
+        ///
+        /// The actual service response to the API is an RFC 7519 JSON Web Token. This token can be retrieved from <see cref="AttestationResponse{T}.Token"/>.
+        /// For the GetPolicy API, the body of the <see cref="AttestationResponse{T}.Token"/> is a <see cref="StoredAttestationPolicy"/> object, NOT a string.
+        /// </remarks>
+        private async Task<AttestationResponse<string>> GetPolicyInternalAsync(AttestationType attestationType, bool async, CancellationToken cancellationToken = default)
         {
             using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(AttestationAdministrationClient)}.{nameof(GetPolicy)}");
             scope.Start();
             try
             {
-                var result = await _policyClient.GetAsync(attestationType, cancellationToken).ConfigureAwait(false);
-                var token = new AttestationToken(result.Value.Token);
+                Response<PolicyResponse> result;
+                if (async)
+                {
+                    result = await _policyClient.GetAsync(attestationType, cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    result = _policyClient.Get(attestationType, cancellationToken);
+                }
+
+                var token = AttestationToken.Deserialize(result.Value.Token, _clientDiagnostics);
                 if (_options.TokenOptions.ValidateToken)
                 {
-                    await token.ValidateTokenAsync(_options.TokenOptions, await GetSignersAsync(cancellationToken).ConfigureAwait(false), cancellationToken).ConfigureAwait(false);
+                    var signers = await GetSignersAsync(async, cancellationToken).ConfigureAwait(false);
+
+                    if (!await token.ValidateTokenInternal(_options.TokenOptions, signers, async, cancellationToken).ConfigureAwait(false))
+                    {
+                        throw new AttestationTokenValidationFailedException("Attestation Token was rejected.")
+                        {
+                            Signers = (IReadOnlyList<AttestationSigner>)signers,
```

Comment:
```suggestion
                            Signers = signers,
```

I may be overlooking something, but from the signature of `GetSignersAsync` on L699,  this should already be typed as the cast.

```csharp
private async Task<IReadOnlyList<AttestationSigner>> GetSignersAsync(bool async, CancellationToken cancellationToken)
```